pkg/api/validation/validation.go

```diff
@@ -1242,6 +1242,29 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldForbidden("data", "Maximum secret size exceeded"))
 	}
 
+	switch secret.Type {
```

Comment:
cut me a tiny bit of slack... I'm trying to keep the service accounts PR balanced on top of this commit :)